testutil/daemon/ops.go

```diff
@@ -17,21 +17,24 @@ func WithDefaultCgroupNamespaceMode(mode string) Option {
 }
 
 // WithTestLogger causes the daemon to log certain actions to the provided test.
-func WithTestLogger(t testing.TB) func(*Daemon) {
+func WithTestLogger(t testing.TB) Option {
 	return func(d *Daemon) {
 		d.log = t
 	}
 }
 
 // WithExperimental sets the daemon in experimental mode
-func WithExperimental(d *Daemon) {
-	d.experimental = true
-	d.init = true
+func WithExperimental() Option {
```

Comment:
This could also be moved into a block of option types, eg.

```go
var (
  WithExperimental Option = withExperimental
  WithInit Option = withInit
)
```

or something similar (where `withExperimental` is the old function so Godoc doesn't try to inline the function body).
This way we wouldn't have to update all the uses since it doesn't take any dependencies anyways. I prefer the way you've done it personally, but just wanted to make sure you've considered the alternative in case you prefer to not have the unnecessary parens every time the option is used.

Reply:
ah, yes, I initially tried `var WithExperimental Option = func(......`, but godoc didn't appear to be documenting `var <somefunc>`, so I decided to use my current approach. I think for consistency the current approach is nicer as well (always `WithSomething()` instead of `WithSomething` for _some_ of the options)